Eplant/views/eFP/Viewer/index.tsx

```diff
@@ -467,4 +469,66 @@ export default class EFPViewer
       {props.geneticElement?.id}
     </Typography>
   )
+
+  citation = (props: { activeData?: EFPViewerData, state?: EFPViewerState, gene?: GeneticElement | null }) => {
+    const [xmlData, setXMLData] = useState<string[]>([])
+
+    const viewID = props.activeData?.views.find((v) => v.id == props.state?.activeView)?.name
+    const viewXML = props.activeData?.views.find((v) => v.id == props.state?.activeView)?.xmlURL
+    useEffect(() => {
+      const xmlLoad = async () => {
+        let xmlString = ""
+        if (viewXML) {
+          try {
+            const response = await fetch(viewXML)
+            const data = await response.text()
+            xmlString = data
+          } catch (error) {
+            console.error('Error fetching xmlData:', error)
+          }
+        }
+
+        // Extract <li> tags
+        if (xmlString !== "") {
+          const parser = new DOMParser();
+          const xmlDoc = parser.parseFromString(xmlString, 'text/xml');
+          const listItems = xmlDoc.querySelectorAll('info li');
+          const itemsArray = Array.from(listItems).map((liElement) => liElement.textContent ? liElement.textContent : "");
+          setXMLData(itemsArray)
+        } else {
+          setXMLData([])
+        }
+      }
+      xmlLoad()
+    }, [viewXML])
+
+    if (viewID) {
+      const citation = getCitation(viewID)
+      // Need to set inner HTML to capture nested HTML tags in some citations
+      return (
+        <div>
+          <div dangerouslySetInnerHTML={{ __html: citation.source }}></div>
+          <br></br><div dangerouslySetInnerHTML={{ __html: citation.notes }}></div>
+          <br></br>{citation.URL ? citation.URL : ""}
```

Comment:
```suggestion
{citation.URL ? <p><a href={citation.URL}>citation.URL</a></p> : ""}
```

Reply:
These should fix the unnecessary spaces that are there now when note/url are empty